app/src/androidTest/java/org/simple/clinic/patient/businessid/BusinessIdMetaAdapterAndroidTest.kt

```diff
@@ -0,0 +1,56 @@
+package org.simple.clinic.patient.businessid
+
+import androidx.test.runner.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.TestClinicApp
+import org.simple.clinic.patient.businessid.BusinessId.MetaVersion.BpPassportMetaVersionV1
+import org.simple.clinic.patient.businessid.BusinessId.MetaVersion.Unknown
+import org.simple.clinic.patient.businessid.BusinessIdMeta.BpPassportMetaV1
+import java.util.UUID
+import javax.inject.Inject
+
+@RunWith(AndroidJUnit4::class)
+class BusinessIdMetaAdapterAndroidTest {
+
+  @field:Inject
+  lateinit var businessIdMetaAdapter: BusinessIdMetaAdapter
+
+  @Before
+  fun setUp() {
+    TestClinicApp.appComponent().inject(this)
+  }
+
+  @Test
+  fun all_known_mappings_of_business_meta_version_must_be_de_serialized_correctly() {
+    // This test is to ensure that whenever a new meta version is added, its corresponding
+    // json adapter is also added to the MoshiBusinessIdMetaAdapter.
+    val businessIdMetas = BusinessId.MetaVersion
+        .values()
+        .map { metaVersion ->
+          metaVersion to when (metaVersion) {
+            is Unknown -> throw RuntimeException("$metaVersion should not be a part of known mappings!")
```

Comment:
Do we need `Unknown` check? It's handled in else block. Also I'm not very comfortable with logic (loops in this case) in tests. Can we look into making this test more straightforward?

Reply:
Discussed offline: This was intentionally added because we don't want to forget to add an adapter for (de)serialising new type of `BusinessIdMeta` objects. The error message is meant to be different for both the `Unknown` case and the other types.